swift/extractor/SwiftExtractor.cpp

```diff
@@ -72,57 +78,48 @@ static fs::path getFilename(swift::ModuleDecl& module, swift::SourceFile* primar
   return resolvePath(filename);
 }
 
-/* The builtin module is special, as it does not publish any top-level declaration
- * It creates (and caches) declarations on demand when a lookup is carried out
- * (see BuiltinUnit in swift/AST/FileUnit.h for the cache details, and getBuiltinValueDecl in
- * swift/AST/Builtins.h for the creation details)
- * As we want to create the Builtin trap file once and for all so that it works for other
- * extraction runs, rather than collecting what we need we pre-populate the builtin trap with
- * what we expect. This list might need thus to be expanded.
- * Notice, that while swift/AST/Builtins.def has a list of builtin symbols, it does not contain
- * all information required to instantiate builtin variants.
- * Other possible approaches:
- * * create one trap per builtin declaration when encountered
- * * expand the list to all possible builtins (of which there are a lot)
- */
-static void getBuiltinDecls(swift::ModuleDecl& builtinModule,
-                            llvm::SmallVector<swift::Decl*>& decls) {
-  llvm::SmallVector<swift::ValueDecl*> values;
-  for (auto symbol : swiftBuiltins) {
-    builtinModule.lookupValue(builtinModule.getASTContext().getIdentifier(symbol),
-                              swift::NLKind::QualifiedLookup, values);
-  }
-  decls.insert(decls.end(), values.begin(), values.end());
-}
-
 static llvm::SmallVector<swift::Decl*> getTopLevelDecls(swift::ModuleDecl& module,
-                                                        swift::SourceFile* primaryFile = nullptr) {
+                                                        swift::SourceFile* primaryFile,
+                                                        const swift::Decl* lazyDeclaration) {
   llvm::SmallVector<swift::Decl*> ret;
+  if (lazyDeclaration) {
+    ret.push_back(const_cast<swift::Decl*>(lazyDeclaration));
```

Comment:
In my defence I can say that Swift compiler doesn't respect const-correctness as well, so we are not making things worse.
Alternative approach, is to drop all the const-ness we have in place for all the decls.

Reply:
another approach, would be to actually return `std::vector<const swift::Decl*>`, at the cost of a slightly more involved way of adding in primary file and module top level decls (something like
```cpp
llvm::SmallVector<swift::Decl*> tmp;
... // fill tmp using original `getTopLevelDecls` from `primaryFile` or `module`
ret.insert(ret.end(), tmp.data(), tmp.data() + tmp.size());
```
I _think_ this could work, but not sure)

Reply:
Good idea!